collector/tcpstat.go

```diff
@@ -0,0 +1,210 @@
+// +build !notcpstat
+
+package collector
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	procTcpStat  = "/proc/net/tcp"
+	procTcp6Stat = "/proc/net/tcp6"
+)
+
+const (
```

Comment:
Cool. So if I understand @brian-brazil correctly, he means he'd prefer seeing a single metric with these as a label dimension. Something like `tcp_connection_states` and a label `states="established|syn_sent|syn_recv|..."`. @brian-brazil Can you confirm?